packages/devtools_app/lib/src/memory/flutter/memory_heap_tree_view.dart

```diff
@@ -537,6 +620,97 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
     );
   }
 
+  bool _classMatcher(HeapGraphClassLive liveClass) {
+    final regExp = RegExp(knownClassesRegExs);
+    return regExp.allMatches(liveClass.name).isNotEmpty;
+  }
+
+  List<ExternalReference> _collect(List dupSearch) {
+    final externalsToAnalyze = <ExternalReference>[];
+
+    // Analyze the external heap for memory information
+    final root = controller.libraryRoot;
+    assert(root != null);
+
+    print(">>>> RegEx = $knownClassesRegExs");
+
+// TODO(terry): Remove dupSearch debug code.
+    for (final library in root.children) {
+      if (library.isExternals) {
+        final ExternalReferences externals = library;
+        for (final ExternalReference external in externals.children) {
+          assert(external.isExternal);
+
+          final liveExternal = external.liveExternal;
+          final size = liveExternal.externalProperty.externalSize;
+          final liveElement = liveExternal.live;
+          final HeapGraphClassLive liveClass = liveElement.theClass;
+          if (_classMatcher(liveClass)) {
+            externalsToAnalyze.add(external);
+          }
+
+          // TODO(terrry): Remove debug code.
+          if (liveClass.name.contains('Image') ||
+              liveClass.name == '_Int32List' ||
+              liveClass.name == 'FrameInfos') {
+            dupSearch.add(liveElement);
+          }
+        }
+      }
+    }
+
+    return externalsToAnalyze;
+  }
+
+  void _analyze() {
+    // TODO(terry): Remove debug code.
+    List debugDupSearch = [];
+
+    final externalsToAnalyze = _collect(debugDupSearch);
+
+    // TODO(terry): Remove this debug code.
+    debugValidateCollecting(externalsToAnalyze, debugDupSearch);
+
+    // Analyze the collected data.
+
+    // TODO(terry): TBD
+    for (final externalRef in externalsToAnalyze) {
+      final liveElement = externalRef.liveExternal.live;
+      final HeapGraphClassLive liveClass = liveElement.theClass;
+      print(liveClass.name);
+      if (liveClass.name == 'ImageCache') {
+        print("STOP");
+      }
+    }
+  }
+
+  // TODO(terry): Remove this debug code.
```

Comment:
fine to leave in this debug code as long as the method name is prefixed with the word debug and the method isn't called in production code.
that matches the sorts of patterns used by flutter reasonably well.

Reply:
Done.